sycl/test-e2e/BFloat16/bfloat16_conversions.cpp

```diff
@@ -55,6 +58,82 @@ template <typename T> int test_host() {
   return 1;
 }
 
+int test_host_vector_conversions() {
+  bool Passed = true;
+  std::cout << "float[4] -> bfloat16[4] -> float[4] conversion on host..."
+            << std::flush;
+
+  float FloatArray[4] = {1.0f, 2.0f, 3.0f, 4.0f};
+
+  // float[4] -> bfloat16[4]
+  bfloat16 BFloatArray[4];
+  sycl::ext::oneapi::detail::FloatVecToBF16Vec<4>(FloatArray, BFloatArray);
+
+  // bfloat16[4] -> float[4]
+  float NewFloatArray[4];
+  sycl::ext::oneapi::detail::BF16VecToFloatVec<4>(BFloatArray, NewFloatArray);
+
+  // Check results.
+  for (int i = 0; i < 4; ++i)
+    Passed &= (FloatArray[i] == NewFloatArray[i]);
+
+  if (Passed)
+    std::cout << "passed\n";
+  else
+    std::cout << "failed\n";
+
+  return !Passed;
+}
+
+int test_device_vector_conversions(queue Q) {
+  int err = 0;
+  buffer<int> err_buf(&err, 1);
+
+  std::cout << "float[4] -> bfloat16[4] conversion on device..." << std::flush;
+  // Convert float array to bfloat16 array
+  Q.submit([&](handler &CGH) {
+     accessor<int, 1, access::mode::write, target::device> ERR(err_buf, CGH);
+     CGH.single_task([=]() {
+       float FloatArray[4] = {1.0f, -1.0f, 0.0f, 2.0f};
+       bfloat16 BF16Array[4];
+       sycl::ext::oneapi::detail::FloatVecToBF16Vec<4>(FloatArray, BF16Array);
+       for (int i = 0; i < 4; i++) {
+         if (FloatArray[i] != (float)BF16Array[i]) {
+           ERR[0] = 1;
+         }
+       }
+     });
+   }).wait();
+
+  if (err)
+    std::cout << "failed\n";
+  else
+    std::cout << "passed\n";
+
+  std::cout << "bfloat16[4] -> float[4] conversion on device..." << std::flush;
+  // Convert bfloat16 array back to float array
+  Q.submit([&](handler &CGH) {
+     accessor<int, 1, access::mode::write, target::device> ERR(err_buf, CGH);
+     CGH.single_task([=]() {
+       bfloat16 BF16Array[3] = {1.0f, 0.0f, -1.0f};
+       float FloatArray[3];
+       sycl::ext::oneapi::detail::BF16VecToFloatVec<4>(BF16Array, FloatArray);
+       for (int i = 0; i < 3; i++) {
+         if (FloatArray[i] != (float)BF16Array[i]) {
+           ERR[0] = 1;
+         }
+       }
+     });
+   }).wait();
+
+  if (err)
+    std::cout << "failed\n";
```

Comment:
if there is a future change in the source that causes this test to fail someday, it'll be more work for whoever catches that ticket.  Make the failure output more explicit.  At minimum output "failed on host" and "failed on device". And if you know of anything else that might be useful to some unlucky co-worker, maybe include that too.

Reply:
Sure. I'll improve the test output.

Reply:
Actually, after looking at the test again, the test will currently output something like:
```
    bfloat16[4] -> float[4] conversion on device...  passed/failed
```
or
```
float[4] -> bfloat16[4] -> float[4] conversion on host.... passed/failed
```
for device and host respectively.  There's a `std::cout` before `queue.submit()` which will output the type of conversion being performed along where the conversion takes place (`host`/`device`).